src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.AbstractRecoverableSyntaxRoot.cs

```diff
@@ -124,10 +124,7 @@ public RecoverableSyntaxRoot<TRoot> WithSyntaxTree(IRecoverableSyntaxTree<TRoot>
 
                 using (var stream = await _storage.ReadStreamAsync(cancellationToken).ConfigureAwait(false))
                 {
-                    var retVal = RecoverRoot(stream, cancellationToken);
-                    _storage.Dispose();
-                    _storage = null;
-                    return retVal;
+                    return RecoverRoot(stream, cancellationToken);
```

Comment:
There is not a right place for disposing.  A solution structure is not disposable.  We've have other services that dispose when the workspace is disposed, though that's probably not what you are looking for because that's often at app shutdown time, and its not a hard requirement to dispose a workspace either.